lib/mongoid/fields/field_types.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Fields
+
+    # Singleton module which contains a mapping of field types to field class.
+    # Custom field types can be configured.
+    #
+    # @api private
+    module FieldTypes
+
+      # The default mapping of field type symbol/string aliases to classes.
+      #
+      # @api private
+      DEFAULT_ALIASES = {
+        array: Array,
+        big_decimal: BigDecimal,
+        binary: BSON::Binary,
+        boolean: Mongoid::Boolean,
+        date: Date,
+        date_time: DateTime,
+        #decimal128: BSON::Decimal128,
+        float: Float,
+        hash: Hash,
+        integer: Integer,
+        #object: Object,
+        object_id: BSON::ObjectId,
+        range: Range,
+        regexp: Regexp,
+        set: Set,
+        string: String,
+        stringified_symbol: Mongoid::StringifiedSymbol,
+        symbol: Symbol,
+        time: Time,
+        #time_with_zone: ActiveSupport::TimeWithZone,
+      }.with_indifferent_access.freeze
+
+      class << self
+
+        # Resolves the user-provided field type to the field type class.
+        #
+        # @example
+        #   Mongoid::FieldTypes.get(:point)
+        #
+        # @param [ Class | Symbol | String | Class ] field_type The field
+        #   type class or its string or symbol alias.
+        #
+        # @return [ Class | nil ] The underlying field type class, or nil if
+        #   string or symbol was passed and it is not mapped to any class.
+        def get(field_type)
+          case field_type
+          when Class
+            type
+          when Symbol, String
+            mapping[field_type]
+          else
+            raise Mongoid::Errors::InvalidFieldType.new(self.name, field, field_type)
```

Comment:
InvalidFieldType's language should be changed to cover the case of invalid type (i.e. an integer instead of symbol/string), UnknownFieldType should be added to replace the current usage of InvalidFieldType when the alias isn't defined.

Reply:
> InvalidFieldType's language should be changed to cover the case of invalid type
If such an error is not actually being called today, we should just rename `InvalidFieldType` to `UnknownFieldType`. After merging this PR, there won't be any error class `InvalidFieldType`. We can add in a future PR if/when it's actually triggered.

Reply:
Resolving as this is outdated.